File: waybar/.config/waybar/scripts/tests/fake_cava.rb
#!/usr/bin/env ruby
# waybar/.config/waybar/scripts/fake_cava.rb

bars = ["▁", "▂", "▃", "▄", "▅", "▆", "▇", "█"]
i = 0

loop do
  s = (0...10).map { |j| bars[(i + j) % bars.length] }.join(" ")
  puts "{\"text\":\"#{s}\"}"
  $stdout.flush
  sleep 0.5
  i = (i + 1) % bars.length
end

File: home/waybar/.config/waybar/scripts/tests/fake_cava.py
# waybar/.config/waybar/scripts/fake_cava.py
#!/usr/bin/env python3
import time, itertools

bars = ["▁", "▂", "▃", "▄", "▅", "▆", "▇", "█"]
for i in itertools.cycle(range(len(bars))):
    s = " ".join(bars[(i + j) % len(bars)] for j in range(10))
    print('{"text":"%s"}' % s, flush=True)
    time.sleep(0.5)
